smack-extensions/src/test/java/org/jivesoftware/smackx/usertune/UserTuneElementTest.java

```diff
@@ -0,0 +1,88 @@
+/**
+ *
+ * Copyright 2019 Aditya Borikar.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.usertune;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.jivesoftware.smack.test.util.XmlUnitUtils.assertXmlSimilar;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.jivesoftware.smack.parsing.SmackParsingException;
+import org.jivesoftware.smack.test.util.SmackTestSuite;
+import org.jivesoftware.smack.test.util.TestUtils;
+import org.jivesoftware.smack.xml.XmlPullParser;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+import org.jivesoftware.smackx.usertune.element.UserTuneElement;
+import org.jivesoftware.smackx.usertune.provider.UserTuneProvider;
+
+import org.junit.Test;
+
+public class UserTuneElementTest extends SmackTestSuite {
+
+    @Test
+    public void toXmlTest() throws IOException, XmlPullParserException, SmackParsingException {
+        String xml = "<tune xmlns='http://jabber.org/protocol/tune'>" +
+                "<artist>Yes</artist>" +
+                "<length>686</length>" +
+                "<rating>8</rating>" +
+                "<source>Yessongs</source>" +
+                "<title>Heart of the Sunrise</title>" +
+                "<track>3</track>" +
+                "<uri>http://www.yesworld.com/lyrics/Fragile.html#9</uri>" +
+                "</tune>";
+
+        URI uri = null;
+        try {
+            uri = new URI("http://www.yesworld.com/lyrics/Fragile.html#9");
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Invalid URI");
```

Comment:
Why do you catch the exception just to throw it again? Simply let the method throw the `URISyntaxException`. This makes the code a lot more readable. Also never swallow Exceptions! With your code we would loose all the information included in `e`.